pages/badges/index.tsx

```diff
@@ -0,0 +1,45 @@
+import type { NextPage } from 'next'
+
+import { Layout } from '@/components/layouts/layout'
+import { SimpleGrid, Button } from '@chakra-ui/react'
+import { useNetwork } from 'wagmi'
+import { getContractAddress } from '@/utils/contractAddress'
+import { useBadges } from '@/hooks/badge/useBadges'
+import { ethers } from 'ethers'
+import { useEffect, useState } from 'react'
+import { BadgeBox } from '@/components/badges/BadgeBox'
+
+const Badges: NextPage = () => {
+  const { activeChain } = useNetwork()
+  const henkakuBadge = getContractAddress({
+    name: 'henkakuBadge',
+    chainId: activeChain?.id
+  })
+  const { badges } = useBadges(henkakuBadge)
+  const [badgeList, setBadgeList] = useState<[]>([])
+
+  useEffect(() => {
+    setBadgeList(badges as [])
+  }, [badges])
+
+  return (
+    <Layout>
+      <SimpleGrid
+        columns={{ sm: 2, md: 4 }}
+        spacing="8"
+        p="10"
+        textAlign="center"
+        rounded="lg"
+      >
+        {badgeList &&
+          badgeList.map((data, index) => (
+            <div key={index}>
+              <BadgeBox badge={data} tokenId={index + 1} />
```

Comment:
I was hoping to be able to use the badge tokenID because the indexes seem to shift in number and order and the paths to the detail pages seem to shift.

Reply:
I thought this was true, so I investigated, but it seems to be quite difficult to return the key (tokenId) of the mapping.

https://github.com/henkaku-center/badge/blob/4707912899d4f6b36024443c6e9caac3750a7077/contracts/HenkakuBadge.sol#L35

I think there are three options.

1. Since the tokenId is guaranteed to be an index, we will leave it as it is.

2.  Put the tokenId in the struct of the contract.

3. instead of using getBagdes, process useBadge in a loop with TokenId on the front end side.

what do you think? @hourglasshoro  @SamuraiT

Reply:
1 or 2. but 1 is ok as you mentioned. it is clear badge is in order

Reply:
I see, if it is guaranteed that tokenId is the same value as index, it seems to be OK.